lib/modules/filteReddit.js

```diff
@@ -311,32 +315,29 @@ module.shouldRun = () => !(
 	module.options.excludeCommentsPage.value && isPageType('comments')
 );
 
-let filterlineKey;
-let filterlineState;
-let initialFilterlineVisibility;
-let filterlineVisibilityOptionKey;
+const pageID = browseContexts.currentLocation.getCurrent();
+const filterlineStorage = Storage.wrap(`RESmodules.filteReddit.${pageID}`, (null: null | *));
+const filterlineVisibilityStorage = Storage.wrap(`RESmodules.filteReddit.filterlineVisibility.${pageID}`, (false: boolean));
 
-module.loadDynamicOptions = () => {
-	const pageID = browseContexts.currentLocation.getCurrent();
-	filterlineKey = `RESmodules.filteReddit.${pageID}`;
-	filterlineVisibilityOptionKey = `filterline_${pageID}`;
+const initialFilterlineState = _.once(() => reifyPromise(filterlineStorage.get()));
+const initialFilterlineVisibility = _.once(() => reifyPromise(asyncSome([
+	filterlineVisibilityStorage.get(),
+	Init.loadOptions.then(() => module.options.showFilterline.value),
+], x => x)));
 
+module.loadDynamicOptions = () => {
 	// Initiate async load now so state will be restored asap
-	filterlineState = Storage.get(filterlineKey);
-
-	module.options[filterlineVisibilityOptionKey] = {
```

Comment:
I shouldn't have suggested this, it doesn't work well when we need multiple keys.

Just loading it with `Storage` in `loadDynamicOptions` is fast enough